encoder-otel-brave/src/main/java/zipkin2/reporter/otel/brave/SpanTranslator.java

```diff
@@ -0,0 +1,215 @@
+    scopeSpanBuilder.addSpans(spanBuilder
+        .build());
+    InstrumentationScope.Builder scopeBuilder = InstrumentationScope.newBuilder();
+    scopeBuilder.setName(BraveScope.getName());
+    scopeBuilder.setVersion(BraveScope.getVersion());
+    scopeSpanBuilder.setScope(scopeBuilder.build());
+    resourceSpansBuilder.addScopeSpans(scopeSpanBuilder
+        .build());
+    tracesDataBuilder.addResourceSpans(resourceSpansBuilder.build());
+    return tracesDataBuilder.build();
+  }
+
+  static KeyValue stringAttribute(String key, String value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setStringValue(value))
+        .build();
+  }
+
+  static KeyValue intAttribute(String key, int value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setIntValue(value))
+        .build();
+  }
+
+  private Span.Builder builderForSingleSpan(MutableSpan span, Builder resourceSpansBuilder) {
+    Span.Builder spanBuilder = Span.newBuilder()
+        .setTraceId(ByteString.fromHex(span.traceId() != null ? span.traceId() : io.opentelemetry.api.trace.SpanContext.getInvalid().getTraceId()))
+        .setSpanId(ByteString.fromHex(span.id() != null ? span.id() : io.opentelemetry.api.trace.SpanContext.getInvalid().getSpanId()))
+        .setName((span.name() == null || span.name().isEmpty()) ? "unknown" : span.name());
+    if (span.parentId() != null) {
+      spanBuilder.setParentSpanId(ByteString.fromHex(span.parentId()));
+    }
+    long start = span.startTimestamp();
+    long finish = span.finishTimestamp();
+    spanBuilder.setStartTimeUnixNano(TimeUnit.MICROSECONDS.toNanos(start));
+    if (start != 0 && finish != 0L) {
+      spanBuilder.setEndTimeUnixNano(TimeUnit.MICROSECONDS.toNanos(finish));
+    }
+    Kind kind = span.kind();
+    if (kind != null) {
+      switch (kind) {
+        case CLIENT:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_CLIENT);
+          break;
+        case SERVER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_SERVER);
+          break;
+        case PRODUCER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_PRODUCER);
+          break;
+        case CONSUMER:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_CONSUMER);
+          break;
+        default:
+          spanBuilder.setKind(SpanKind.SPAN_KIND_INTERNAL); //TODO: Should it work like this?
+      }
+    }
+    else {
+      spanBuilder.setKind(SpanKind.SPAN_KIND_INTERNAL); //TODO: Should it work like this?
+    }
+    String localServiceName = span.localServiceName();
+    if (localServiceName == null) {
+      localServiceName = Resource.getDefault().getAttribute(ServiceAttributes.SERVICE_NAME);
+    }
+    resourceSpansBuilder.getResourceBuilder()
+        .addAttributes(stringAttribute(ServiceAttributes.SERVICE_NAME.getKey(), localServiceName));
+    String localIp = span.localIp();
```

Comment:
you can probably see a pattern here, where we check if a string is null then add attribute.


consider maybeAddStringAttribute(key, nullableVal) and similar

Reply:
Fixed in https://github.com/openzipkin-contrib/zipkin-otel/pull/9/commits/b7868dd0ac5ebd8822b7851435923d061a221a99